corehq/apps/data_interfaces/models.py

```diff
@@ -1214,6 +1324,101 @@ def to_dict(self):
         }
 
 
+class CaseDuplicateNew(models.Model):
+    id = models.BigAutoField(primary_key=True)
+    case_id = models.CharField(max_length=126, db_index=True)
+    action = models.ForeignKey("CaseDeduplicationActionDefinition", on_delete=models.CASCADE)
+    match_values = models.CharField(max_length=256)
+
+    class Meta:
+        db_table = "data_interfaces_caseduplicate_new"
+        unique_together = ('case_id', 'action')
+        indexes = [
+            models.Index(fields=['match_values', 'action_id'])
+        ]
+
+    def __str__(self):
+        return (
+            f"CaseDuplicateNew("
+            f"case_id={self.case_id}, action_id={self.action_id}, match_values={self.match_values})"
+        )
+
+    def delete(self, *args, check_for_orphans=True, **kwargs):
+        with transaction.atomic():
+            if check_for_orphans:
+                other_records = CaseDuplicateNew.objects.filter(
+                    action=self.action, match_values=self.match_values).exclude(case_id=self.case_id)[:2]
+
+                if other_records.count() == 1:
+                    # This will be orphaned when the current record is deleted, so delete it as well
+                    other_records[0].delete(check_for_orphans=False)
+
+            return super().delete(*args, **kwargs)
+
+    @classmethod
+    def create(cls, case, action, save=True):
```

Comment:
Do you think it could be confusing to have two different "create" methods?
```py
CaseDuplicateNew.create(...)
CaseDuplicateNew.objects.create(...)
```
Could it make sense to override the manager's `create` method instead?

Reply:
Do you feel strongly about this? I was following the example mentioned in [the django docs](https://docs.djangoproject.com/en/5.0/ref/models/instances/#django.db.models.Model), where the intention was to override the `__init__` method (but that created issues).  Defaulting `save` to `True` was a nod to the name overlap creating an expectation that any `create` method would save to the database, but the intended use case was actually to support creating this object without hitting the database. Manager methods, to me, are intended to work with the database. So, it felt confusing to put a method designed to support avoiding the database on a custom manager. I also don't love the indirection when trying to look up code capabilities -- it feels harder to discover this functionality, because it isn't on the base class.

Reply:
It's interesting that the django docs list the preferred method second:

> 2. Add a method on a custom manager (usually preferred):

I don't feel that strongly about it.